### PR TITLE
Add forms for post, comment and like.

### PR DESCRIPTION
**Pull request by chudisoft**
[Project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/forms_project_v1_1.md#project-requirements)

    Create a method called current_user in your ApplicationController to make current user data available to all controllers.
        It will return the first user from the database.
        Don't worry about proper authentication at this time. You will add it in the upcoming projects.
    Create forms to perform the following functions:
        Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
        Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
    Allow Users to add likes to Posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
-  root 'pages#hello'
-  # get 'pages/hello'
+  resources :users
+  get '/', to: 'users#index', as: 'home'
+  # get '/users', to: 'users#index', as: 'users'
+  # get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/new' , to: 'posts#new', as: 'new_user_post'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  get '/users/:user_id/posts/:id/like', to: 'likes#create', as: 'user_post_like'
```
Comment (DammyShittu) [inline config/routes.rb]: - [ ] Kindly use the correct route for the like creation. The like should be wrapped within the posts and users routes, so just like `:user_id`, there should be `:post_id`

- [ ] The like create should be a `post` request not `get` request

  ```
  post `/users/:user_id/posts/:post_id/likes`, ...
  ```

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+
+
+  get '/users/:user_id/posts/:id/comments/new' , to: 'comments#new', as: 'new_user_post_comment'
+  post '/users/:user_id/posts/:id/comments/create' , to: 'comments#create', as: 'user_post_comments'
```
Comment (DammyShittu) [inline config/routes.rb]: - [ ] The comments routes should use `:post_id`, not `:id`.

- [ ] For the route on line 13, there is no use for `/create`. The post create route should end at the `/comments` word.

  ```
  post '/users/:user_id/posts/:post_id/comments', ...
  ```

Comment (DammyShittu) [inline config/routes.rb]: - **[OPTIONAL]** All these routes can be defined more conventionally and shortly.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
  ```

  When you run `rails routes` in the terminal, this is what the routes should look like.

  image

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,17 @@
 Rails.application.routes.draw do
-  root 'pages#hello'
-  # get 'pages/hello'
+  resources :users
+  get '/', to: 'users#index', as: 'home'
+  # get '/users', to: 'users#index', as: 'users'
+  # get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/new' , to: 'posts#new', as: 'new_user_post'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```
Comment (DammyShittu) [inline config/routes.rb]: - [ ] Kindly create the route for post create action

File: README.md [hunk 4]
```diff
@@ -33,11 +29,12 @@
```
Comment (DammyShittu) [inline README.md:29]: - [ ] Kindly remove this migration file:

  ```
  db\migrate\20231026234155_add_indeces.rb
  ```

  It prevents your project from being set up and raises duplicate index error. You are defining indexes there but, those indexes already exist before getting to that file. Notice that you have indexes in the migration files already.

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  # Create action to handle post creation
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to @post, notice: 'Post was successfully created.'
+    else
+      render 'new'
+    end
+  end
+
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb]: - [ ] The post should be created using the `current_user` method you defined in the application controller.

File: app/views/posts/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<h2>Create a New Post</h2>
+<%= form_for Post.new do |f| %>
```
Comment (DammyShittu) [inline app/views/posts/new.html.erb]: - [ ] There is an error when I visit this page, so I cannot create post from the browser. Kindly fix it.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @chudisoft,

## Highlights
✔️ Like and comment creation feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @chudisoft, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implemented the features ✅ 
- Nice *UI* design ✅ 
image


## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**